test/spec/request-test.js

```diff
@@ -1425,4 +1427,263 @@ describe('Request-Promise', function () {
+                    'Age': 20
+                }
+                content: {
+                    name: 'resource4'
+                }
+            }
+        ]
+
+        var incrementResource = function(resource) {
+            resource.header['Last-Modified'].add(5, 'seconds')
+        }
+
+        before(function (done) {
+            // This creates a local cache server to test cache behavior of the lib
+            cacheServer = http.createServer(function (request, response) {
+                (bodyParser.json())(request, response, function () {
+                    var path = url.parse(request.url).pathname;
+                    var resourceName = parseInt(path.split('?')[0].split('/')[1]);
+                    var resourceIndex = resourceName[ resourceName.length - 1 ]
+                    var resource = resources[resourceIndex]
+                    if (resource){
+                        response.writeHead(200, resource.header);
+                        response.end(JSON.stringify(resource.content));
+                    } else {
+                        response.writeHead(404)
+                        reponse.end('')
+                    }
+                });
+            });
+
+            cacheServer.listen(5000, function () {
+                done();
+            });
+
+        });
+
+        after(function () {
+            cacheServer.close();
+        });
+
+        beforeEach(function () {
+            // require a fresh lib
+            delete require.cache[require.resolve('../../lib/rp.js')]
+            rp = require('../../lib/rp.js')
+
+            // reset the last modified of the resources
+            for (i=0; i<resources.length; i++){
+                resources[i].headers['Last-Modified'] = moment().format(LAST_MODIFIED_FORMAT)
+            }
+        });
+
+        assertResponse = function(response, statusCode, content, lastModified, age) {
+            expect(response.statuCode).to.be.equal(statusCode)
+            expect(JSON.parse(response.body).name).to.be.equal(content)
+            expect(response.headers['last-modified']).to.be.equal(now)
+            if (!age) {
+                expect(response.headers['age']).to.be.equal(age)
+            } else {
+                expect(response.headers['age'] > 0).to.be.true
+            }
+        }
+
+        it('should cache resource : cache-control, cache option true', function (done) {
+            var options = {
+                uri: 'http://localhost:5000/resource1',
+                method: 'GET',
+                cache: true
+            }
+
+            rp(options).then(function(response){
+                assertResponse(response, 200, 'resource1', now, 0)
+                rp(options).then(function(response){
+                    assertResponse(response, 200, 'resource1', now, true)
+                    done()
+                })
+            })
+
+        });
+
+        it('should not cache resource : no cache-control, cache option true', function (done) {
```

Comment:
Test case

Assumptions: 
- the resource returns no `cache-control`
- the cache is previously empty
- the cache option is true

results:
- the resource does not get cache after first request
- the resource that we get after second request is a fresh one